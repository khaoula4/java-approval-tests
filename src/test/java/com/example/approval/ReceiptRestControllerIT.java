package com.example.approval;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.servlet.MockMvc;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;

@SpringBootTest
@AutoConfigureMockMvc
public class ReceiptRestControllerIT {


    @Autowired
    private MockMvc mockMvc;

    @Test
    void shouldReturnApprovedReceiptTextFromApi() throws Exception {
        String actualResponseBody = mockMvc.perform(get("/api/receipts/1001"))
                .andReturn()
                .getResponse()
                .getContentAsString();

        Approvals.verify(actualResponseBody);
    }
}
