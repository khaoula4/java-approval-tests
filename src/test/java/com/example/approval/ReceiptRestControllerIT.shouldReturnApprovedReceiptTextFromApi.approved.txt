RECEIPT FOR ORDER 1001 :

Tea x2 @ 1,50 = 3,00
Coffee x1 @ 3,20 = 3,20
Cookie x3 @ 0,80 = 2,40

TOTAL = 8,60
